ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/CommandStore.java

```diff
@@ -139,6 +150,8 @@ public QueuedCommandStatus enqueueCommand(
    * @return The commands that have been polled from the command topic
    */
   public List<QueuedCommand> getNewCommands() {
+    completeSatisfiedOffsetFutures();
```

Comment:
The assumption being made here is that all the QueuedCommands returned by the last call to getNewCommands have been handled, and therefore its safe to complete their futures. A more explicit way of doing this would be to return the offsets in QueuedCommand, and then have the caller acknowledge them explicitly. I think that would clear the confusion here.

Reply:
As I understand it, the current code just gets the current consumer position and uses that to complete the futures. Which is totally valid. 

The answer to my original question, i.e. can it be done _after_ the poll, is 'no'. Because that would mean we'd be completing futures for the list of `QueuedCommand`s the function is about to return, and they haven't been actioned yet.   So, for me, this is resolved.